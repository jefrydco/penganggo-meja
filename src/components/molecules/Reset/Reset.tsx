import { Button, ButtonProps } from "@mui/material";
import React from "react";

export interface ResetProps {
  onClick?: ButtonProps['onClick']
}

export const Reset: React.FC<ResetProps> = ({ onClick }) => {
  return <Button role="button" variant="contained" onClick={onClick}>Reset</Button>
};
