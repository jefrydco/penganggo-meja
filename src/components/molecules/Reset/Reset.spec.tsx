import React from 'react'
import { describe, expect, it, vi } from 'vitest'
import { render, userEvent } from '@/utils/tests/vitest'
import { Reset } from './Reset'

describe('Reset', () => {
  it('match snapshot', () => {
    const { asFragment } = render(<Reset />)
    expect(asFragment()).toMatchSnapshot()
  })
  it.todo('clickable', () => {
    const handleClick = vi.fn()
    const { getByRole } = render(<Reset onClick={handleClick} />)
    userEvent.click(getByRole('button'))
    expect(handleClick).toBeCalledTimes(1)
  })
})
