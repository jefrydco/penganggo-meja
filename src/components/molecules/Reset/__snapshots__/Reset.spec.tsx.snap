// Vitest Snapshot v1

exports[`Reset > match snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
    role="button"
    tabindex="0"
    type="button"
  >
    Reset
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
